### PR TITLE
Add cypress tests to test Basic example

### PR DESCRIPTION
**Pull request by ankri**
This PR adds several cypress tests to test the basic example

- `frame.ts` - In here we test that dragging around components inside the `<Frame />` is possible
- `history.ts` - In here we test that we can undo and redo changes made to the Editor
- `props.ts` - In here we test that we can change the props via the SettingsPanel
- `toolbox.ts` - In here we test that we can drag components from the Toolbox into the Frame

Additionally I added `data-cy` attributes to the components inside the basic example (and modified them a tiny bit) to make them easier to test with cypress.

I also added two new commands to `package.json`: `cypress:open` to run cypress with the UI and `cypress.run` to run cypress in headless mode.

## Test videos
It's kinda hard to follow the tests in the videos since they are so fast, but it should give a sneak peek into what is being tested.

### `frame.ts`
https://user-images.githubusercontent.com/2842920/108121958-da564880-70a3-11eb-8577-1745a2d6c341.mp4

### `history.ts`
https://user-images.githubusercontent.com/2842920/108121447-1fc64600-70a3-11eb-8991-06a7f4d5aa5e.mp4

### `props.ts`
https://user-images.githubusercontent.com/2842920/108121490-2c4a9e80-70a3-11eb-88c4-9451fb98967e.mp4

### `toolbox.ts`
https://user-images.githubusercontent.com/2842920/108121507-32407f80-70a3-11eb-9044-479372481f9a.mp4

Thank you @prevwong for providing me with the drag and drop commands. I took the code and modified it a bit. All those commands can be found inside `commands.ts`.

My plan was to provide a reference implementation for other developers that plan to test their craft-based applications. That's why I kept the tests so verbose and added so many comments. I hope that makes the tests easy to read and easy to follow.

Next steps:
- Integrate the tests into the CI Pipeline
- Publish a `@craftjs/cypress` package with the craft-specific commands from `commands.ts` so other users can import it in their tests and get access to the commands

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,6 @@ lib
 .now
 .vscode
 .idea
+.DS_Store
+cypress/videos/*
+cypress/screenshots/*
```
Comment (prevwong) [inline .gitignore:14]: Add a newline here

File: cypress.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "baseUrl": "http://localhost:3002"
+}
```
Comment (prevwong) [inline cypress.json:3]: Newline here too

File: cypress/support/commands.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      return (
+        // the next.js render indicator shares some of the styles but additionally has an id
+        element.id === '' &&
+        element.classList.length === 0 &&
+        element.style.position === 'fixed' &&
+        element.style.zIndex === '99999' &&
+        element.style.backgroundColor === indicatorColors[indicator]
+      );
+    });
+  }
```
Comment (prevwong) [inline cypress/support/commands.ts:70]: Can we move all the dnd-related commands (ie: DummyDataTransfer, getDropIndicator, drop etc) into a separate file; maybe `dnd.ts` and then just import that here?

**Review by prevwong — CHANGES_REQUESTED** [review]
Looks really good 😍 Just some minor changes and we're golden!

File: cypress/support/commands.ts (hunk 3 above)
Comment (ankri) [reply cypress/support/commands.ts]: Done. In phase 2 I want to publish the actions in `dnd.ts` as a separate `@craft/cypress` package :)

**Review by prevwong — APPROVED** [review]
